src/board/UBBoardView.cpp

```diff
@@ -1492,29 +1494,74 @@ void UBBoardView::mouseDoubleClickEvent (QMouseEvent *event)
     mousePressEvent (event);
 }
 
+// Mouse Wheel Zoom
+// Source: https://wiki.qt.io/Smooth_Zoom_In_QGraphicsView
 void UBBoardView::wheelEvent (QWheelEvent *wheelEvent)
 {
-    QList<QGraphicsItem *> selItemsList = scene()->selectedItems();
-    // if NO have selected items, than no need process mouse wheel. just exist
-    if( selItemsList.count() > 0 )
+
+    if( wheelEvent->modifiers() & Qt::ControlModifier )
+    {
+        wheelZoomEvent(wheelEvent);
+    }
+    else
     {
-        // only one selected item possible, so we will work with first item only
-        QGraphicsItem * selItem = selItemsList[0];
 
-        // get items list under mouse cursor
-        QPointF scenePos = mapToScene(wheelEvent->pos());
-        QList<QGraphicsItem *> itemsList = scene()->items(scenePos);
 
-        bool isSelectedAndMouseHower = itemsList.contains(selItem);
-        if(isSelectedAndMouseHower)
+        QList<QGraphicsItem *> selItemsList = scene()->selectedItems();
+        // if NO have selected items, than no need process mouse wheel. just exist
+        if( selItemsList.count() > 0 )
         {
-            QGraphicsView::wheelEvent(wheelEvent);
-            wheelEvent->accept();
-        }
+            // only one selected item possible, so we will work with first item only
+            QGraphicsItem * selItem = selItemsList[0];
 
+            // get items list under mouse cursor
+            QPointF scenePos = mapToScene(wheelEvent->pos());
+            QList<QGraphicsItem *> itemsList = scene()->items(scenePos);
+
+            bool isSelectedAndMouseHower = itemsList.contains(selItem);
+            if(isSelectedAndMouseHower)
+            {
+                QGraphicsView::wheelEvent(wheelEvent);
+                wheelEvent->accept();
+            }
+
+        }
     }
+}
+
+void UBBoardView::wheelZoomEvent(QWheelEvent *event){
+    int numDegrees = event->delta() / 8;
+     int numSteps = numDegrees / 15; // see QWheelEvent documentation
+     _numScheduledScalings += numSteps;
+     if (_numScheduledScalings * numSteps < 0) // if user moved the wheel in another direction, we reset previously scheduled scalings
+     _numScheduledScalings = numSteps;
+
 
+
+     QTimeLine *anim = new QTimeLine(150, this);
+     anim->setCurveShape(QTimeLine::LinearCurve);
+     anim->setUpdateInterval(10);
```

Comment:
An update interval of 10ms is quite fast. In most cases the default of 40ms should be sufficient to create a smooth animation. Note that repainting may be expensive if you have many complicated objects on the screen.